modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/DefaultOperationContext.java

```diff
@@ -59,9 +62,30 @@ public DefaultOperationContext(ConfigurationInstance<Object> configuration, Reso
      * {@inheritDoc}
      */
     @Override
-    public Object getParameter(String parameterName)
+    public <T> T getParameter(String parameterName)
     {
-        return parameters.get(parameterName);
+        return (T) parameters.get(parameterName);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public <T> T getTypeSafeParameter(String parameterName, Class<? extends T> expectedType)
+    {
+        T value = getParameter(parameterName);
```

Comment:
I need to get the return value in my hands so that i can construct the pretty error message. Doing it as you say would prevent that